npm/angular/src/mount.ts

```diff
@@ -230,6 +229,10 @@ function setupComponent<T> (
     })
   }
 
+  if (component.ngOnChanges) {
+    component.ngOnChanges()
```

Comment:
My Angular is a little rusty, but doesn't `ngOnChanges` expect to receive an object with the prev/new value for all the changed props? Wouldn't we need to construct an appropriate arg for this function?

Also, the original issue showed that components mounted using template strings *were* properly calling `ngOnChanges` - would this result in a double-call in those situations?

Reply:
Agree we should include a parameter that matches the `ngOnChanges` input. Here is how [atl](https://github.com/testing-library/angular-testing-library/blob/36648c4b431a9e17efa1f59cc2d83710c17a42ee/projects/testing-library/src/lib/testing-library.ts#L291) is doing it, we don't support `rerender` so it doesn't have to be too fancy.

@mike-plummer the component created via the use of templates will never have an `ngOnChanges` hook so no need to worry about the double call.

Reply:
@ZachJW34 I may be misunderstanding, but I don't think that's quite accurate. The template string example provided in the [original issue](https://github.com/cypress-io/cypress/issues/23591) *does* have an `ngOnChanges` that is properly firing ATM, so I think this change would cause it to double-fire on mount. Were you referring to the fact that you shouldn't expect it to fire *after* initial mount since there's no way to update the props?

Reply:
If I'm not mistaken, `ngOnChanges` only fires when a data-bound input property changes. So I think we should only call it when `componentProperties` is set.

Reply:
@mike-plummer I should have been more clear, when using a template, the component passed in via the template is rendered via a Proxy component called `WrapperComponent`. Then reason `ngOnChanges` works with the template is due to the fact that changing properties on the ProxyComponent (set via `componentProperties`) on first mount gets detected by change detection, and due to the template binding calls the child components `ngOnChanges` lifecycle event.

So there wouldn't be a double fire because the explicit call of `ngOnChanges` for template examples would be trying to fire the Proxy component's `ngOnChanges` which is a noop since it won't be defined.